### PR TITLE
[RSDK-6874][RSDK-7279]: Rover making endless noise after calculating control parameters & fix tuning

### PR DESCRIPTION
**Pull request by martha-johnston**
I think this should work but I'll test tomorrow in office to confirm.

### CODE REVIEW

**Review by JohnN193 — APPROVED** [review]

**Review by JohnN193 — COMMENTED** [review]
main concern is if a user tries to move the rover, they could [trigger tuning again](https://github.com/viamrobotics/rdk/blob/c724fa9cc230b12386db4a512cc2ed1d3082eafe/components/motor/gpio/motor_encoder.go#L381) if they call GoFor(I think)

Also do we need to handle sensor controlled tuning too?

**Comment by martha-johnston** [comment]
Yes calling GoFor immediately after would trigger tuning again, but I _think_ that would also happen in the current set up. I don't think it should be an issue because they need to copy the pid values over to the config anyways which will cause it to reconfigure. 

And sensor controlled base is already stopping the loop each time at the end of `tuneSinglePID`

**Comment by JohnN193** [comment]
gotcha, I just recall not being able to move the rover via set power after tuning, but I wasn't sure if that would still be the case with this PR

**Comment by martha-johnston** [comment]
> gotcha, I just recall not being able to move the rover via set power after tuning, but I wasn't sure if that would still be the case with this PR

I'm not aware of this issue can you elaborate? you can't use SetPower?

File: components/base/sensorcontrolled/velocities.go [hunk 1]
```diff
@@ -114,7 +114,7 @@ func (sb *sensorBase) SetState(ctx context.Context, state []*control.Signal) err
 	sb.mu.Lock()
 	defer sb.mu.Unlock()
 
-	if !sb.loop.Running() {
+	if sb.loop != nil && !sb.loop.Running() {
```
Comment (martha-johnston) [inline components/base/sensorcontrolled/velocities.go:117]: okay tuning was broken added fix to this PR

**Comment by martha-johnston** [comment]
> > gotcha, I just recall not being able to move the rover via set power after tuning, but I wasn't sure if that would still be the case with this PR
> 
> I'm not aware of this issue can you elaborate? you can't use SetPower?

Update: just tried to use SetPower after tuning and before updating the config values and it worked, so I'm assuming this got fixed somewhere along the way in the refactoring

**Pull request merged**